### ISSUE TITLE
spectral widget does not show velocities for CASA images

### ISSUE THREAD

**Issue by jott3077** [CARTAvis/carta-frontend#1935]
Loading a CASA cube into CARTA. The spectral viewer shows channels, or when WCS is chosen, frequency. No velocity is displayed. This is likely due to the velocity as an additional axis label for spectral CASA cubes and frequency is native. 
By default, the CASA wcs should be the velocity label if it exists. 

The following could be split as a separate ticket, also some of it have been discussed in other places:

A more general approach would be to be able to select a range of options for any format, this includes channel, frequency, velocity (could be multiple axes for CASA images)  in different systems (minimum LSRK/BARY and radio/optical), wavelengths. 

A nice thing would be to add a box where rest frequencies can be overwritten.

**Comment by kswang1029** [comment]
@jott3077 
Right now x-axis is determined by CTYPE3 and no conversion is applied.
We are hoping we can enhance the spectral profiler widget to some extent in v1.3. Please refer to https://docs.google.com/document/d/10nG54sC8XoEKhlGNnTmY6ZOHstF3zeiJi22gBjEvE_k/edit#

**Comment by veggiesaurus** [comment]
so should the `CTYPE3` be velocity by default for _CASA_ images?

**Comment by kswang1029** [comment]
I tried with a test CASA image and the axes are
`'axisnames': array(['Right Ascension', 'Declination', 'Frequency', 'Stokes']`

If we convert a CASA image to FITS with exportfits task in CASA, the default is frequency. There is an option to set CTYPE3 in velocity. 
`velocity            =      False        #  Use velocity (rather than frequency) as spectral axis`

**Comment by jott3077** [comment]
yes, the default is frequency but with velocity=True in exportfits, the axis of the fits image will be velocity. Usually this is what the users want if a velocity label in the CASA image exists. 

So yes, I would suggest to always use the velocity for CTYPE if it exists. (I guess for images with Stokes it will be CTYPE4 not CTYPE3, given the CASA convention to list Stokes before the spectral axis - that is unless the fits converter swaps the axes)

**Comment by jott3077** [comment]
fyi I see the same problem for miriad images

**Comment by kswang1029** [comment]
@jott3077 @veggiesaurus we do have spectral conversion via AST. For this issue, maybe we could add a toggle in the preferences such as "display velocity when possible" so that we will show velocity as the default axis in the spectral profiler widget. If so, this issue should be transferred to frontend repo.

**Comment by kswang1029** [comment]
> @jott3077 @veggiesaurus we do have spectral conversion via AST. For this issue, maybe we could add a toggle in the preferences such as "display velocity when possible" so that we will show velocity as the default axis in the spectral profiler widget. If so, this issue should be transferred to frontend repo.

Any more input regarding this? Shall we transfer this to the frontend repo?

**Comment by jott3077** [comment]
this sounds good to me. The velocity is the most important to be displayed and it would be inconvenient to always switch over from frequency. For spectral line surveys there should not be a rest frequency attached in the first place so the display will be in frequency. But velocity should be the default display if available.
